### ISSUE TITLE
Unsupported Accept header, try: application/json

### ISSUE THREAD

**Issue by john-qin** [begriffs/postgrest#274]
I have latest postgrest and postgresql installed on ubuntu 14.04.  when I test it,  I am getting " Unsupported Accept header, try: application/json".  how do I fix it ?

**Comment by begriffs** [comment]
Try including the request header
```HTTP
Accept: application/json
```
:smile: 
If you're hitting the route with a web browser it is probably asking for HTML and PostgREST is saying it doesn't respond with HTML. It can do `application/json` or `text/csv`.

To customize request headers you'll need to use a rest client like [Postman](https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop?hl=en) or [curl](http://curl.haxx.se/).

**Comment by begriffs** [comment]
Also you can join us in the [chat room](https://gitter.im/begriffs/postgrest) and I'll walk you through how to make requests with the right headers.

**Comment by PierreR** [comment]
`curl` by default is sending `Accept: */*` and I would bet the browsers do the same. In that case, I would think `postgrest` should send `json` as the default format because the client is basically saying it will deal with whatever.

**Comment by NikolayS** [comment]
Actually, PostgREST isn't intended to be called directly from browser. It's API for programs, not humans :)
You can try following: 
 - if you prefer terminal tests, add `-H 'Accept:application/json'` to `curl` call (writing some shell script that will simplify tests and allow you to skip all redundant curl options, as well as calling perl via pipeline – is obvious)
 - if you have Firefox, use Network tab provided in its dev tools. Open that tab, then make a call via inserting your URL calling your API to address bar, catch `Unsupported Accept header` and then you'll see the call in Network tab. All you need then is to edit `Accept` header using "Edit and Resend" button. It's located on the right, in "Headers" subtab:
![screenshot 2015-09-01 01 39 16](https://cloud.githubusercontent.com/assets/1345402/9592281/5af342c0-504a-11e5-9982-8fbe2e5c1c49.png)
- if you don't feel comfortable with this, and have Chrome, use this extension with clear and pretty interface: https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop?hl=en

Hope this helps.

**Comment by NikolayS** [comment]
BTW, I am +1 for returning json by default, with `Accept: */*` is provided

**Comment by begriffs** [comment]
Ah you're right, I overlooked the `*/*` at the end of the long accept header that chrome sends. Postgrest should definitely notice that and serve json.

**Comment by john-qin** [comment]
why am I still getting this error message ?  I set the content type to application/json

{"readyState":4,"responseText":"Unsupported Accept header, try: application/json","status":415,"statusText":"Unsupported Media Type"}

I am using jquery.

            $("#btnSubmit").click(function () {

                var title = $("#tbxTitle").val()
                var description = $("#txtDescription").val()
                var rating = $("#rating").val()
                var created_date = new Date()

                var request = { "title": title, "description": description, "category": "iPhone", "rating": rating, "created_date": created_date }

                $.ajax({
                    url: "http://localhost:3000/app",
                    type: "POST",
                    data: JSON.stringify(request),
                    contentType: 'application/json; charset=utf-8',
                    success: function (data) {
                        alert('Inserted Successfully');
                        alert(JSON.stringify(data))
                    },
                    error: function (msg) {
                        //alert("error:")
                        alert(JSON.stringify(msg));
                    }
                });
            })

**Comment by begriffs** [comment]
I'll fix the problem where it is disregarding `*/*`, but until then I think you can add the necessary header with something like

```javascript
            $.ajax({
                url: "http://localhost:3000/app",
                type: "POST",
                data: JSON.stringify(request),
                headers: { 
                    Accept: "application/json; charset=utf-8"
                },
                success: function (data) {
                    alert('Inserted Successfully');
                    alert(JSON.stringify(data))
                },
                error: function (msg) {
                    //alert("error:")
                    alert(JSON.stringify(msg));
                }
            });
```

**Comment by begriffs** [comment]
The fix is now released in 0.2.11.1, and active on the demo server.

**Comment by PierreR** [comment]
Thanks for the quick fix and release. I have tested it and it works like a charm.
